### PR TITLE
Chncy 006/login signup page

### PR DESCRIPTION
**Pull request by Hakusen-git**
**Issue:**
Create basic sign in and sign up pages

**Solution:**
Using AWS webpackage to achieve most of the functionality.
Using AWS lambda and API gateway to manage data into database

**Risk:**
Log out functionality needs to be done soon.
Bookmark functionality needs to be implemented

**Reviewed By:**
Steven Ho

### CODE REVIEW

File: src/pages/password.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+          onSuccess: function(data){
+            console.log(data);
+            handleSuccess(true);
+          },
+          onFailure: function(err) {
+            alert(err.message || JSON.stringify(err));
+          },
+          inputVerificationCode: function(data) {
+            handleSuccess(true);
+            console.log("Input code: ", data);
```
Comment (kcho9906) [inline src/pages/password.js]: remove console logs

File: src/pages/login.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        if (err) {
+            alert(err);
+        }
+      });  
+
+      history.push("/revise");
+
+      
+
+        // AWS.config.credentials = new AWS.CognitoIdentityCredentials({
```
Comment (kcho9906) [inline src/pages/login.js]: whats this commented code for?

**Review by kcho9906 — CHANGES_REQUESTED** [review]

File: src/pages/password.js (hunk 1 above)
Comment (Hakusen-git) [reply src/pages/password.js]: Removed in the latest commit

File: src/pages/login.js (hunk 2 above)
Comment (Hakusen-git) [reply src/pages/login.js]: Hitarth wrote this code so I think he can explain it better. If he thinks it's not necessary I can remove them

File: src/api/userApi.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        //   .then(result => console.log(result));
+
+        await fetch(`${API_ROOT}/users/post/`, {
+            method: 'POST',
+            body: JSON.stringify(user)
+        }).then(result => result.json());
+    } catch (error) {
+        console.log(error);
+    }
+};
```
Comment (kcho9906) [inline src/api/userApi.js]: missing new line

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "files.exclude": {
+    "amplify/.config": true,
+    "amplify/**/*-parameters.json": true,
+    "amplify/**/amplify.state": true,
+    "amplify/**/transform.conf.json": true,
+    "amplify/#current-cloud-backend": true,
+    "amplify/backend/amplify-meta.json": true,
+    "amplify/backend/awscloudformation": true
+  }
+}
```
Comment (kcho9906) [inline .vscode/settings.json:11]: .vscode should be in gitignore

File: amplify/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Getting Started with Amplify CLI
+This directory was generated by [Amplify CLI](https://docs.amplify.aws/cli).
+
+Helpful resources:
+- Amplify documentation: https://docs.amplify.aws
+- Amplify CLI documentation: https://docs.amplify.aws/cli
+- More details on this folder & generated files: https://docs.amplify.aws/cli/reference/files
+- Join Amplify's community: https://amplify.aws/community/
```
Comment (kcho9906) [inline amplify/README.md:8]: Can delete this file

File: src/api/userApi.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      method: "POST",
+      body: JSON.stringify(user),
+    })
+      .then((result) => result.json())
+      .then((result) => console.log(result));
+
+    // await fetch(`${API_ROOT}/users/post/`, {
+    //   method: "POST",
+    //   body: JSON.stringify(user),
+    // }).then((result) => result.json());
```
Comment (kcho9906) [inline src/api/userApi.js:17]: can remove comment

File: src/components/AccountMenu.js [hunk 7]
```diff
@@ -1,81 +1,100 @@
...
 import ListItemIcon from "@material-ui/core/ListItemIcon";
 import ListItemText from "@material-ui/core/ListItemText";
-import ExitToAppIcon from '@material-ui/icons/ExitToApp';
-import SettingsIcon from '@material-ui/icons/Settings';
-import ToggleOffIcon from '@material-ui/icons/ToggleOff';
-import ToggleOnIcon from '@material-ui/icons/ToggleOn';
+import ExitToAppIcon from "@material-ui/icons/ExitToApp";
+import SettingsIcon from "@material-ui/icons/Settings";
+import ToggleOffIcon from "@material-ui/icons/ToggleOff";
+import ToggleOnIcon from "@material-ui/icons/ToggleOn";
```
Comment (kcho9906) [inline src/components/AccountMenu.js]: Nit pick but I think importing like this is bad - just do 
```import { ExitToAppIcon, SettingsIcon, ToggleOffIcon } from "@material-ui/icons"``` would work I think

**Review by kcho9906 — CHANGES_REQUESTED** [review]

**Review by kcho9906 — APPROVED** [review]

**Pull request merged**